src/reason_pprint_ast.ml

```diff
@@ -3089,7 +3096,12 @@ class printer  ()= object(self:'self)
           | Ppat_lazy p ->formatPrecedence (label ~space:true (atom "lazy") (self#simple_pattern p))
           | Ppat_extension e -> self#extension e
           | Ppat_exception p ->
-              makeList ~postSpace:true [atom "exception"; self#pattern p]
+              (*
+                An exception pattern should be wrapped in (...)
+                Example: (Sys_error _ as exc)
+                  results in incorrect parsing with type error otherwise.
+              *)
+              makeList ~postSpace:true [atom "exception"; self#pattern ~wrap:true p]
```

Comment:
Oh, we already have a helper for generating a "wrapped" pattern. It's called `self#simple_pattern p`.